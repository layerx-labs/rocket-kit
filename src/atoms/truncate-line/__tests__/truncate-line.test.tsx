import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import TruncateLine from '..';

describe('Truncate Line', () => {
  it('renders', () => {
    const { asFragment } = render(
      <TruncateLine value="this is my huge text" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('has correct value', () => {
    const textValue =
      'Lorem ipsum, or lipsum as it is sometimes known, is dummy text used in laying out print, graphic or web designs. The passage is attributed to an unknown typesetter in the 15th century who is thought to have scrambled parts of Cicero De Finibus Bonorum et Malorum for use in a type specimen book.';
    const testId: string = 'truncate-test';
    render(<TruncateLine value={textValue} dataTestId={testId} />);
    expect(screen.getByText(textValue)).toBeInTheDocument();
  });
});
