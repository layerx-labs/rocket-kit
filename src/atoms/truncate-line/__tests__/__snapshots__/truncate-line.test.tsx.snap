// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Truncate Line renders 1`] = `
<DocumentFragment>
  .c0 {
  display: table;
  table-layout: fixed;
  width: 100%;
  white-space: nowrap;
}

.c0 span {
  display: table-cell;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
    class="c0 truncate"
  >
    <span>
      this is my huge text
    </span>
  </div>
</DocumentFragment>
`;
